Model/ModelManager.php

```diff
@@ -376,7 +386,7 @@ public function getNormalizedIdentifier($entity)
         }
 
         // the entities is not managed
-        if (!$entity /*|| !$this->getEntityManager($entity)->getUnitOfWork()->isInIdentityMap($entity) // commented for perfomance concern */) {
+        if (!$entity || !$this->getEntityManager($entity)->getUnitOfWork()->isInIdentityMap($entity)) {
```

Comment:
yes, `AbstractAdmin::update()` calls `ModelManager::update()` which calls `flush()`. In our project we then have all manner of doctrine events firing at that time, so it gets complicated.

Stepping back a bit, why do we care here if the object is in the UOW? Couldn't we use UnitOfWork::getEntityState() instead if some kind of state check is needed? I'm not sure of the case that @acrobat is trying to catch here, but I think all we care about is "can we get an identifier from the entity in hand", right?

If there's other logic related to detecting whether an object is new or not then I'm not sure it lives in `ModelManager::getNormalizedIdentifier()` whose purpose in life is just to "Get the identifiers for this model class as a string." So, imho, returning null just because we can't find the object in the UOW (even though the object is in the database and the ModelManager can work out an id for it) is not the best position to be in.